gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java

```diff
@@ -59,22 +90,78 @@ protected String getLogPrefix() {
     return DEFAULT_LOG_PREFIX;
   }
 
+  public List<ImageConfiguration> customizeConfig(List<ImageConfiguration> configs) {
+    kitLogger.info("Running in %s mode", kubernetesExtension.getRuntimeMode().getLabel());
+    if (kubernetesExtension.getRuntimeMode() != RuntimeMode.OPENSHIFT) {
+      kitLogger.info("Building Docker image in Kubernetes mode");
+    }
+    // TODO: Run Generators
+    return  configs;
+  }
+
   protected JKubeServiceHub.JKubeServiceHubBuilder initJKubeServiceHubBuilder() {
-    return JKubeServiceHub.builder()
-        .log(kitLogger)
-        .configuration(JKubeConfiguration.builder()
-            .project(javaProject)
-            .reactorProjects(Collections.singletonList(javaProject))
-            .build())
-        .clusterAccess(clusterAccess)
-        .offline(kubernetesExtension.getOffline().getOrElse(DEFAULT_OFFLINE))
-        .platformMode(kubernetesExtension.getRuntimeMode());
+    JKubeServiceHub.JKubeServiceHubBuilder builder = JKubeServiceHub.builder()
+            .log(kitLogger)
+            .configuration(JKubeConfiguration.builder()
+                    .project(javaProject)
+                    .reactorProjects(Collections.singletonList(javaProject))
+                    .sourceDirectory(kubernetesExtension.getSourceDirectory().getOrElse("src/main/docker"))
+                    .outputDirectory(kubernetesExtension.getOutputDirectory().getOrElse("build/docker"))
+                    .registryConfig(RegistryConfig.builder()
+                            .settings(Collections.emptyList())
+                            .authConfig(kubernetesExtension.authConfig != null ? kubernetesExtension.authConfig.toMap() : null)
+                            .skipExtendedAuth(kubernetesExtension.getSkipExtendedAuth().getOrElse(false))
+                            .passwordDecryptionMethod(s -> s)
+                            .registry(kubernetesExtension.getPullRegistry().getOrElse(kubernetesExtension.getRegistry().getOrElse("docker.io")))
+                            .build())
+                    .build())
+                    .clusterAccess(clusterAccess)
+                    .offline(kubernetesExtension.getOffline().getOrElse(DEFAULT_OFFLINE))
+                    .platformMode(kubernetesExtension.getRuntimeMode());
+    if (isDockerAccessRequired()) {
```

Comment:
Would evaluating the content of this if condition be a problem in case there was no effective Docker access?